Cargo.toml

```diff
@@ -9,3 +9,9 @@ members = [
 default-members = [
     "src/wrappers/themis/rust",
 ]
+
+# Freeze `rayon` version (used by `criterion` used by benchmarks)
+# so that benchmarks still build/run with Rust 1.58.
+# FIXME: remove this after we bump minimum required Rust version
+[patch.crates-io]
+rayon = { git = "https://github.com/rayon-rs/rayon.git", tag = "v1.6.1" }
```

Comment:
You can just add `rayon = "=1.6.1"` to `[dev-dependencies]` alongside `criterion` and Cargo would select it, even if rayon's not actually used by our code.

The way this works, Cargo uses a single version of a dependency everywhere if it is possible to satisfy all version constraints with it. So if one of the crates in the workspace demands 1.6.1 exactly, that will be used for all crates, as long as they use version specs that would allow 1.6.1 (e.g., `1`, `^1.3`, etc.)

Since we don't publish benchmarks to crates.io, this is probably okay as it is. But holding back a published dependency is not an idiomatic use case for `[patch]`. Though, it is convenient in way, since now you don't have to update every individual crate with a "fake" dependency to make `-p` work.

Reply:
Thanks, good point, I forgot about such possibility. Switched to this method. Also, had to explicitly add `rayon-core` (which was the main issue here) because otherwise saw `cargo tree` showing selected version of `rayon` but latest (1.11) for `rayon-core`